lib/json-schema/validator.rb

```diff
@@ -238,6 +242,46 @@ def validation_errors
       @errors
     end
 
+    def resolve_nested_references(schema, schema_portion, list)
```

Comment:
Actually I think there is no better name. Values coming in here as ```schema_portion``` are either Hashes, Arrays or primitive values and they are all nested parts of the schema to be resolved (they are neither all hashes nor all valid schemas on their own).